RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -92,28 +92,35 @@ MuonHLTSeedMVAClassifier::MuonHLTSeedMVAClassifier(const edm::ParameterSet& iCon
       l1MuonToken_(consumes<l1t::MuonBxCollection>(iConfig.getParameter<edm::InputTag>("L1Muon"))),
       l2MuonToken_(consumes<reco::RecoChargedCandidateCollection>(iConfig.getParameter<edm::InputTag>("L2Muon"))),
       trackerGeometryToken_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()),
-
       rejectAll_(iConfig.getParameter<bool>("rejectAll")),
       isFromL1_(iConfig.getParameter<bool>("isFromL1")),
-
-      mvaFileB_(iConfig.getParameter<edm::FileInPath>(isFromL1_ ? "mvaFileBL1" : "mvaFileBL2")),
-      mvaFileE_(iConfig.getParameter<edm::FileInPath>(isFromL1_ ? "mvaFileEL1" : "mvaFileEL2")),
-
-      mvaScaleMeanB_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleMeanBL1" : "mvaScaleMeanBL2")),
-      mvaScaleStdB_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleStdBL1" : "mvaScaleStdBL2")),
-      mvaScaleMeanE_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleMeanEL1" : "mvaScaleMeanEL2")),
-      mvaScaleStdE_(iConfig.getParameter<std::vector<double>>(isFromL1_ ? "mvaScaleStdEL1" : "mvaScaleStdEL2")),
-
+      mvaFileB_(iConfig.getParameter<edm::FileInPath>("mvaFileB")),
+      mvaFileE_(iConfig.getParameter<edm::FileInPath>("mvaFileE")),
+      mvaScaleMeanB_(iConfig.getParameter<std::vector<double>>("mvaScaleMeanB")),
+      mvaScaleStdB_(iConfig.getParameter<std::vector<double>>("mvaScaleStdB")),
+      mvaScaleMeanE_(iConfig.getParameter<std::vector<double>>("mvaScaleMeanE")),
+      mvaScaleStdE_(iConfig.getParameter<std::vector<double>>("mvaScaleStdE")),
       doSort_(iConfig.getParameter<bool>("doSort")),
       nSeedsMaxB_(iConfig.getParameter<int>("nSeedsMaxB")),
       nSeedsMaxE_(iConfig.getParameter<int>("nSeedsMaxE")),
-
       etaEdge_(iConfig.getParameter<double>("etaEdge")),
       mvaCutB_(iConfig.getParameter<double>("mvaCutB")),
       mvaCutE_(iConfig.getParameter<double>("mvaCutE")),
-
       minL1Qual_(iConfig.getParameter<int>("minL1Qual")),
       baseScore_(iConfig.getParameter<double>("baseScore")) {
+  const auto& mvaFileBPath = mvaFileB_.fullPath();
+  const auto& mvaFileEPath = mvaFileE_.fullPath();
+
+  if (!isFromL1_ and
+      ((mvaFileBPath.find("FromL1") != std::string::npos) or (mvaFileEPath.find("FromL1") != std::string::npos))) {
+    throw cms::Exception("ConfigurationError")
+        << " isFromL1 parameter is False, but using FromL1 MVA files.\n Please check your configuration";
+  } else if (isFromL1_ and ((mvaFileBPath.find("FromL1") == std::string::npos) or
+                            (mvaFileEPath.find("FromL1") == std::string::npos))) {
+    throw cms::Exception("ConfigurationError")
+        << " isFromL1 parameter is True, but not using FromL1 MVA files.\n Please check your configuration";
+  }
+
```

Comment:
Sorry, but I find this hard-coding of syntax of expected filenames depending on other parameter values (isFromL1) somewhat of an overkill. This module should not enforce some syntax set elsewhere.

Reply:
From [CMSHLT-3197](https://its.cern.ch/jira/browse/CMSHLT-3197)

> then there should probably be a check throwing an exception in case the MVA model is not compatible with the value of "isFromL1", if that makes sense

Reply:
Well, this is a check on the filename, which is still not checking whether the content of the named file is adequate.

Reply:
> which is still not checking whether the content of the named file is adequate.

I am not sure one call tell from the file content. So far the convention has been respected, I don't see why we should break it now.

Reply:
But now it is there, so fine.

Reply:
> I am not sure one call tell from the file content. 

Ok, it looks like one can tell from the number of variables in the xml:

https://github.com/cms-sw/cmssw/blob/f1b02548e1dfa1269f96ce117a39857e8d50cd89/RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc#L33

vs

https://github.com/cms-sw/cmssw/blob/f1b02548e1dfa1269f96ce117a39857e8d50cd89/RecoMuon/TrackerSeedGenerator/src/SeedMvaEstimator.cc#L37

let me see if I can improve this.

Reply:
https://github.com/cms-sw/cmssw/pull/45063/commits/361c1ce2cad720c20c2249b98ad76dd57591c28c should address it.